internal/services/firewall/firewall_policy_application_rule_collection_resource.go

```diff
@@ -0,0 +1,432 @@
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+								ValidateFunc: validation.Any(
+									validation.IsIPAddress,
+									validation.IsCIDR,
+									validation.StringInSlice([]string{`*`}, false),
+								),
+							},
+						},
+						"source_ip_groups": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"destination_addresses": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+								ValidateFunc: validation.Any(
+									validation.IsIPAddress,
+									validation.IsCIDR,
+									validation.StringInSlice([]string{`*`}, false),
+								),
+							},
+						},
+						"destination_fqdns": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"destination_urls": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"destination_fqdn_tags": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"terminate_tls": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+						"web_categories": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceFirewallPolicyApplicationRuleCollectionCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Firewall.FirewallPolicyRuleGroupClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	ruleCollectionGroupId, err := parse.FirewallPolicyRuleCollectionGroupID(d.Get("rule_collection_group_id").(string))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, ruleCollectionGroupId.RuleCollectionGroupName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("checking for existing Firewall Policy Rule Collection Group %q (Resource Group %q / Policy %q): %+v", ruleCollectionGroupId.RuleCollectionGroupName, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, err)
```

Comment:
This can be simplified to:

```suggestion
			return fmt.Errorf("checking for existing Firewall Policy Rule Collection Group %q: %+v", ruleCollectionGroupId, err)
```